veracruz-utils/src/policy.rs

```diff
@@ -137,21 +145,46 @@ pub struct VeracruzIdentity {
     roles: Vec<VeracruzRole>,
 }
 
-impl VeracruzIdentity {
-    /// Creates a new identity from a certificate, and identifier, and a mix
-    /// of Veracruz roles.
+impl<U> VeracruzIdentity<U> {
```

Comment:
What's the advantage of adding the type abstraction here?

Reply:
The combination of client ID, certificate, and roles is used in a few places in Veracruz (e.g. `Baja` and `veracruz-utils`).  However, depending on where the type is used, the actual representation of the certificate is different: in Baja the certificate is represented as the `Certificate` type from RusTLS, whereas in veracruz-utils it is represented as an unparsed string that was read from the global policy.  By introducing the type-abstraction, we can use a single type for representing the client ID, certificate, roles combination whilst being ambivalent about the exact representation of the certificate.

Reply:
Thanks, I see that now. I didn't see the other uses. Nice.